tsdb/tsm1/engine_schema.go

```diff
@@ -0,0 +1,256 @@
+package tsm1
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/cursors"
+	"github.com/influxdata/influxql"
+)
+
+// TagValues returns an iterator which enumerates the values for the specific
+// tagKey in the given bucket matching the predicate within the
+// time range (start, end].
+func (e *Engine) TagValues(ctx context.Context, orgID, bucketID influxdb.ID, tagKey string, start, end int64, predicate influxql.Expr) (cursors.StringIterator, error) {
+	encoded := tsdb.EncodeName(orgID, bucketID)
+
+	var tagKeyBytes []byte
+	if tagKey == "_measurement" {
```

Comment:
I don't think the strings `_measurement` or `_field` should appear within the `tsm1` package. The conversion should probably happen in the `storage/reads` or somewhere similar to that.

Reply:
Agree – it is expected that the correct tokens are passed at this point. 

@jacobmarble we will need to make sure query is aware. They already do this in the `reads` package when transforming a Flux expression (note `models.MeasurementTagKey`):

https://github.com/influxdata/influxdb/blob/2909927ab0eaa2943905f5beab5bbb8d71ca18b8/storage/reads/predicate.go#L268-L274

Reply:
Removed these references